fireplace/controller.py

```diff
@@ -0,0 +1,59 @@
+import asyncio
+from hearthstone.enums import Step
+from fireplace.ai.player import BaseAI
+from fireplace.exceptions import GameOver
+from fireplace.logging import get_logger
+
+
+class GameController:
+	def __init__(self, game):
+		self.log = get_logger("fireplace")
+		self.game = game
+		self.game.manager.register(self)
+		self.previous_step = Step.INVALID
+
+	# manager interface
+	def action_start(self, type, source, index, target):
+		pass
+
+	def action_end(self, type, source):
+		pass
+
+	def new_entity(self, entity):
+		pass
+
+	def start_game(self):
+		self.log.info("Game has started")
+		# note: choice will be None here but won't be when mulligan() starts
+		if isinstance(self.game.player1, BaseAI):
+			self.event_loop.call_soon(self.game.player1.mulligan)
+		if isinstance(self.game.player2, BaseAI):
+			self.event_loop.call_soon(self.game.player2.mulligan)
+
+	def game_step(self, step, next_step):
+		self.log.debug("Game.STEP changes to %s", step)
+		if step == Step.MAIN_ACTION and self.previous_step == Step.MAIN_START:
+			self.log.info("Turn %s starting for player %s", self.game.turn, self.game.current_player.entity_id - 1)
```

Comment:
Use `%i` for turn; `%r` and `self.game.current_player` for the player. (this is a nasty way to get the player id)